app/models/miq_db_config.rb

```diff
@@ -1,221 +0,0 @@
-      "internal"
-    else
-      "external_evm"
-    end
-  end
-
-  def save
-    valid = self.valid?(:from_save => true)
-    return @errors unless valid == true
-
-    _log.info("Validation was successful, saving new settings: #{options.merge(@@pwd_mask).inspect}")
-    vmdb_config = save_without_verify
-    MiqRegion.sync_with_db_region(vmdb_config.config)
-    true
-  end
-
-  def save_without_verify
-    @bkup_ext = Time.now.strftime("%Y%m%d-%H%M%S")
-    save_method = "save_#{@options[:name]}"
-    save_method = "save_common" unless self.respond_to?(save_method)
-    send(save_method)
-  end
-
-  def save_internal
-    self.class.backup_file(@@db_yml, @bkup_ext)
-    save_common
-  end
-
-  def save_common
-    current = VMDB::Config.new("database")
-    self.class.backup_file(@@db_yml, @bkup_ext)
-
-    [:development, :production].each do |env|
-      current.config[env] = @@defaults.merge(@options).delete_if { |n, _v| n == :name }
-    end
-    current.save_file(@@db_yml)
-    current.update_cache_metadata
-    current
-  end
-
-  def valid?(options = {})
-    valid = validate_common
-    return @errors unless valid == true
-
-    valid = verify_config(options[:from_save])
-    valid ? valid : @errors
-  end
-
-  def validate; self.valid?; end
-
-  def validate_common
-    return true if @options[:name] == "internal"
-
-    @errors = ActiveModel::Errors.new(self)
-    valid = true
-    @@common_options.each do |fld|
-      next if fld[:required] != true
-      if @options[fld[:name]].blank?
-        @errors.add(:field, "'#{fld[:description]}' is required")
-        valid = false
-        next
-      end
-
-      if fld[:numeric] == true && !is_numeric?(@options[fld[:name]])
-        @errors.add(:field, "'#{fld[:description]}' must be a numeric")
-        valid = false
-      end
-    end
-    valid
-  end
-
-  def self.backup_file(file, ext)
-    bkup = "#{file}.#{ext}"
-    File.delete(bkup) if File.exist?(bkup)
-    require 'fileutils'
-    FileUtils.copy(file, bkup)
-  end
-
-  def self.human_attribute_name(attribute_key_name, _options = {}); attribute_key_name.to_s.humanize; end
-
-  def method_missing(m, *args)
-    if m.to_s.ends_with?("=")
-      attr = m.to_s.split("=").first.to_sym
-      mode = :set
-    else
-      attr = m
-      mode = :get
-    end
-
-    unless @@common_fields.include?(attr) || attr == :adapter
-      super
-      return
-    end
-
-    attr = :name if attr == :adapter
-
-    case mode
-    when :get then return @options[attr]
-    when :set then @options[attr] = *args
-    end
-  end
-
-  def verify_config(from_save = nil)
-    @errors = ActiveModel::Errors.new(self)
-
-    with_temporary_connection do |conn|
-      tables = conn.tables
-
-      # If we're not saving, return only if we were able to issue a query or not
-      return !!tables unless from_save
-
-      if tables.empty?
-        _log.info("No migrations have been run... migrating")
-        ActiveRecord::Tasks::DatabaseTasks.migrate
```

Comment:
Ok, chatted with @Fryguy...

The original use cases in this method were for the OpsUI configure database option where we wanted to point to a database and it would figure out if it needed to be create/migrate/or yell at you if the database had pending migrations.

Since the time of that code, the appliance_console added options to configure the database, where it would ask you if you wanted to create a database region or join an existing one.  In the case of create, it's pretty clear, it blows away the existing database and sets it up using the provided region number.  In the case of joining an existing region database, it was doing more than it really needed to or should do.

We only really want configure ourselves to use that database by setting the database.yml and synchronize the REGION file to match that DB's region. We should never migrate a database we're joining as other code on other appliances is using that database.